plat/hisilicon/hikey960/hikey960_pm.c

```diff
@@ -295,6 +296,14 @@ static const plat_psci_ops_t hikey960_psci_ops = {
 int plat_setup_psci_ops(uintptr_t sec_entrypoint,
 			const plat_psci_ops_t **psci_ops)
 {
+	unsigned int id;
+
+	hikey960_read_boardid(&id);
+	if (id == 5300)
```

Comment:
Static analysis error - MISRA C-2012 Rule 10.4 violation: The operands "id" and "5300" have essential type categories "unsigned 32-bit int" and "signed 16-bit int", which do not match.

Suggest - if (id == 5300U)